heim-host/src/os/windows.rs

```diff
@@ -1,11 +1,17 @@
 //! Windows-specific extensions.
 
 use std::net::IpAddr;
+use winapi::um::winnt::PSID;
+
+use heim_common::Result;
 
 /// Extension for [User] struct.
 ///
 /// [User]: ../../struct.User.html
-pub trait UserExt {
+pub trait UserExt: Sized {
+    #[doc(hidden)]
+    fn try_from_sid(sid: PSID) -> Result<Self>;
```

Comment:
As far as I understand, `PSID` is a raw pointer to `SID` struct.
It seems safer to use `&SID` in here